consensus/service/src/validators.rs

```diff
@@ -175,41 +154,12 @@ impl<L: Ledger + Sync> TxManagerUntrustedInterfaces for DefaultTxManagerUntruste
 #[cfg(test)]
 pub mod well_formed_tests {
     use super::*;
-    use mc_common::logger::{bench_with_logger, test_with_logger, Logger};
-    use mc_ledger_db::LedgerDB;
-    use mc_transaction_core::{
-        constants::MAX_TOMBSTONE_BLOCKS, ring_signature::KeyImage, tx::Tx,
-        validation::TransactionValidationError,
-    };
+    use mc_common::logger::{test_with_logger, Logger};
     use mc_transaction_core_test_utils::{
         create_ledger, create_transaction, initialize_ledger, AccountKey,
     };
     use rand::SeedableRng;
     use rand_hc::Hc128Rng;
-    use test::Bencher;
-
-    fn is_well_formed(tx: &Tx, ledger: &LedgerDB) -> TransactionValidationResult<()> {
-        let mut rng = Hc128Rng::from_seed([77u8; 32]);
-
-        let untrusted = DefaultTxManagerUntrustedInterfaces::new(ledger.clone());
-
-        let key_images: Vec<KeyImage> = tx.key_images();
-        let membership_proof_highest_indices = tx.get_membership_proof_highest_indices();
-        let output_public_keys = tx.output_public_keys();
-
-        let (cur_block_index, membership_proofs) = untrusted.well_formed_check(
-            &membership_proof_highest_indices[..],
-            &key_images[..],
-            &output_public_keys[..],
-        )?;
-
-        mc_transaction_core::validation::validate(
```

Comment:
This requires full access to the transaction, and so must be done within the enclave.